File: source-base/sep.backend.v1/Data/Entities/ClassApplication.cs
namespace sep.backend.v1.Data.Entities
{
    public class ClassApplication : BaseEntity
    {
        public int Id { get; set; }
        public int PupilId { get; set; }
        public int SemesterId { get; set; }
        public string Title { get; set; }
        public string Description { get; set; }
        public int ApplicationCategoryId { get; set; }
        public string? Response { get; set; }
        public int Status { get; set; }       
        public virtual Pupil? Pupil { get; set; }
        public virtual Semester? Semester { get; set; }
        public virtual ClassApplicationCategory? ClassApplicationCategory { get; set; }
    }
}

File: source-base/source.base.fe/src/types/model/class-application.ts
export interface ResponseGetClassApplication {
  id: number;
  pupilId: number;
  firstName: string;
  lastName: string;
  donorName?: string;
  semesterId: number;
  title: string;
  description: string;
  applicationCategoryId: number;
  categoryName?: string;
  response?: string;
  status: number;
  statusName?: string;
  createDate: Date;
}

export interface RequestGetCategory {
  id: number;
  name: string;
}
export interface RequestResponseClassApplication {
  id : number;
  response : string;
  status : number;
}
export interface ErrorResponseClassApplication {
  Id?  : string[];
  Response? : string[];
  Status? : string[];
}

export interface ErrorCreateAndUpdateClassApplication {
  PupilId?: string[];
  SemesterId?: string[];
  Title?: string[];
  Description?: string[];
  ApplicationCategoryId?: string[];
}

export enum ApplicationStatus {
  PENDING = 1,
  APPROVED = 2,
  REJECTED = 3,
}

export enum ApplicationCategory {
  Default = 1,
}